presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTileFunctions.java

```diff
@@ -0,0 +1,314 @@
+    @ScalarFunction("point_to_bing_tile")
+    @SqlType(BingTileType.NAME)
+    public static long pointToBingTile(
+            @SqlType(GEOMETRY_TYPE_NAME) Slice point,
+            @SqlType(StandardTypes.BIGINT) long zoomLevel)
+    {
+        OGCGeometry geometry = deserialize(point);
+
+        checkCondition(geometry.geometryType().equals(POINT), "point_to_bing_tile expects a Point, but got %s", geometry.geometryType());
+        checkZoomLevel(zoomLevel);
+
+        OGCPoint ogcPoint = (OGCPoint) geometry;
+        checkCondition(!ogcPoint.isEmpty(), "Point must not be empty");
+
+        final double longitude = ogcPoint.X();
+        final double latitude = ogcPoint.Y();
+
+        checkCondition(validLatitude(latitude), LATITUDE_OUT_OF_RANGE);
+        checkCondition(validLongitude(longitude), LONGITUDE_OUT_OF_RANGE);
+
+        return latitudeLongitudeToTile(latitude, longitude, (int) zoomLevel).encode();
+    }
+
+    @Description("Given a Bing tile, returns polygon representation of the tile")
+    @ScalarFunction("bing_tile_polygon")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice bingTilePolygon(@SqlType(BingTileType.NAME) long input)
+    {
+        BingTile tile = BingTile.decode(input);
+
+        return serialize(tileToPolygon(tile));
+    }
+
+    @Description("Given a geometry and a zoom level, returns a minimum set of Bing tiles that fully cover that geometry")
+    @ScalarFunction("geometry_to_bing_tiles")
+    @SqlType("array(" + BingTileType.NAME + ")")
+    public static Block geometryToBingTiles(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(StandardTypes.BIGINT) long zoomLevel)
+    {
+        checkZoomLevel(zoomLevel);
+
+        OGCGeometry geometry = deserialize(input);
+        checkCondition(!geometry.isEmpty(), "Input geometry must not be empty");
+
+        Envelope envelope = new Envelope();
+        geometry.getEsriGeometry().queryEnvelope(envelope);
+
+        checkCondition(validLatitude(envelope.getYMin()) && validLatitude(envelope.getYMax()), "Latitude span for the geometry must be in [%.2f, %.2f] range", MIN_LATITUDE, MAX_LATITUDE);
+        checkCondition(validLongitude(envelope.getXMin()) && validLongitude(envelope.getXMax()), "Longitude span for the geometry must be in [%.2f, %.2f] range", MIN_LONGITUDE, MAX_LONGITUDE);
+
+        boolean boundingBox = isBoundingBox(geometry, envelope);
+
+        BingTile leftUpperTile = latitudeLongitudeToTile(envelope.getYMax(), envelope.getXMin(), (int) zoomLevel);
+        BingTile rightLowerTile = latitudeLongitudeToTile(envelope.getYMin(), envelope.getXMax(), (int) zoomLevel);
+
+        // XY coordinates start at (0,0) in the left upper corner and increase left to right and top to bottom
+        long tileCnt = (rightLowerTile.getX() - leftUpperTile.getX() + 1) * (rightLowerTile.getY() - leftUpperTile.getY() + 1);
+
+        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), (int) tileCnt);
```

Comment:
you can declare `tileCount` as `int` above, also you may want to use `toIntExact `.